tamr_client/transformations.py

```diff
@@ -0,0 +1,150 @@
+import requests
+from simplejson.errors import JSONDecodeError
+
+from tamr_client import dataset, response
+from tamr_client._types import Instance, JsonDict, Project, Session
+from tamr_client._types.transformations import InputTransformation, Transformations
+from tamr_client.exception import TamrClientException
+
+
+class InvalidInputDataset(TamrClientException):
+    """Raised when a Dataset within a InputTransformation
+    is not an input dataset of the target project."""
+
+    pass
+
+
+class LintingError(TamrClientException):
```

Comment:
Stylistically, our exception names omit words like "exception" or "error" since those are implied by them being exceptions in the first place, but not sure if I can think of a non-awkward wording otherwise. Would something slightly more specific like `LintingFailed` make sense here?

E.g. instead of `AlreadyExistsError` we just say `AlreadyExists`

Not a big deal if we can't come up with something better.